controllers/tokenupload_controller.go

```diff
@@ -167,6 +161,13 @@ func (r *TokenUploadReconciler) reconcileRemoteSecret(ctx context.Context, uploa
 		auditLog.Info("manual secret partial update completed")
 	} else {
 		auditLog.Info("manual secret upload initiated", "action", "UPDATE")
+
+		err = remoteSecret.ValidateUploadSecret(uploadSecret)
+		if err != nil {
+			auditLog.Info("manual secret upload failed because of invalid upload secret")
```

Comment:
Maybe do the validation before we output the auditlog entry, because at that point we've not yet done anything audit-worthy..

Reply:
done, thanks